### PR TITLE
Change the return type of softmax function to Status

### PR DESCRIPTION
**Pull request by PeixuanZuo**
### Description
Change the return type of Softmax function(`dispatch_warpwise_softmax_forward `and `dispatch_blockwise_softmax_forward`) from `void ` to `Status`.

### Motivation and Context
Softmax function will call TunableOp which return Status. It's necessary to pass the `Status` from inner function to outer function.

### CODE REVIEW

File: onnxruntime/contrib_ops/cuda/transformers/generation_device_helper.cc [hunk 1]
```diff
@@ -337,12 +337,15 @@ Status ProcessLogits(const OrtValue& logits,                                 //
 
   const CudaT* X_data = is_reuse_logits_buffer ? logits_data : reinterpret_cast<const CudaT*>(next_token_logits.data());
 
-  dispatch_blockwise_softmax_forward<CudaT, float, float, true>(
+  auto status = dispatch_blockwise_softmax_forward<CudaT, float, float, true>(
```
Comment (cloudhan) [inline onnxruntime/contrib_ops/cuda/transformers/generation_device_helper.cc:340]: ORT_RETURN_IF_ERROR

File: onnxruntime/contrib_ops/cuda/transformers/sampling_cuda_helper.h [hunk 2]
```diff
@@ -87,14 +87,17 @@ Status Sample(AllocatorPtr& allocator,
...
+  Status status;
   gsl::span<float>& d_sorted_softmaxed_score = sampling_state->d_sorted_softmaxed_score;
-  dispatch_blockwise_softmax_forward<CudaT, float, float, false>(cuda_stream,
-                                                                 d_sorted_softmaxed_score.data(),
-                                                                 reinterpret_cast<CudaT*>(d_sorted_score.data()),
-                                                                 parameters->vocab_size,
-                                                                 parameters->vocab_size,
-                                                                 parameters->vocab_size,
-                                                                 parameters->batch_size);
+  status = dispatch_blockwise_softmax_forward<CudaT, float, float, false>(cuda_stream,
```
Comment (cloudhan) [inline onnxruntime/contrib_ops/cuda/transformers/sampling_cuda_helper.h:92]: ORT_RETURN_IF_ERROR

File: onnxruntime/contrib_ops/cuda/transformers/sampling_cuda_helper.h [hunk 3]
```diff
@@ -122,13 +125,16 @@ Status Sample(AllocatorPtr& allocator,
...
 
   gsl::span<float>& d_softmaxed_score = sampling_state->d_softmaxed_score;
-  dispatch_blockwise_softmax_forward<CudaT, float, float, false>(cuda_stream,
-                                                                 d_softmaxed_score.data(),
-                                                                 reinterpret_cast<CudaT*>(next_token_scores.data()),
-                                                                 parameters->vocab_size,
-                                                                 parameters->vocab_size,
-                                                                 parameters->vocab_size,
-                                                                 parameters->batch_size);
+  status = dispatch_blockwise_softmax_forward<CudaT, float, float, false>(cuda_stream,
```
Comment (cloudhan) [inline onnxruntime/contrib_ops/cuda/transformers/sampling_cuda_helper.h:128]: Same here and below

**Comment by cloudhan** [comment]
Otherwise, it looks generally good.

File: onnxruntime/contrib_ops/cuda/transformers/generation_device_helper.cc (hunk 1 above)
Comment (PeixuanZuo) [reply onnxruntime/contrib_ops/cuda/transformers/generation_device_helper.cc]: Done.

File: onnxruntime/contrib_ops/cuda/transformers/sampling_cuda_helper.h (hunk 2 above)
Comment (PeixuanZuo) [reply onnxruntime/contrib_ops/cuda/transformers/sampling_cuda_helper.h]: Done.

File: onnxruntime/contrib_ops/cuda/transformers/sampling_cuda_helper.h (hunk 3 above)
Comment (PeixuanZuo) [reply onnxruntime/contrib_ops/cuda/transformers/sampling_cuda_helper.h]: Done.

**Review by cloudhan — DISMISSED** [review]

**Review by cloudhan — APPROVED** [review]
LGTM
